Code/autopkglib/MunkiInstallsItemsCreator.py

```diff
@@ -89,6 +89,11 @@ def create_installs_items(self):
         # Get pkginfo from output plist.
         pkginfo = plistlib.loads(out)
         installs_array = pkginfo.get("installs", [])
+        
+        # Raise if installs_array is empty
+        if not installs_array:
+            raise ProcessorError(f"cannot generate installs_array from: faux_root {faux_root}, "
```

Comment:
Is there a more descriptive error we could provide? What should an admin do if they see this? What did they do wrong and how can they fix it? I'd prefer something that indicates at least where to look or what actions to take to try and fix it.

Reply:
I attempted this with that, but maybe too verbose and not clear?

`Error in com.github.dataJAR-recipes.munki.Audiomovers Listento: Processor: MunkiInstallsItemsCreator: Error: cannot generate installs_array from: faux_root /Users/ben/Library/AutoPkg/Cache/com.github.dataJAR-recipes.munki.Audiomovers Listento, installs_item_paths: ['/Applications/Listento.app']`

Would this benefit from maybe a link to a wiki page giving more details or something?

Reply:
What should someone do if they see this message?

Reply:
2 things:

1. Remove the processor from the recipe, if it's not needed (there will be cases where it's been added, hasn't returned anything and the recipe has been written to overcome but the processor step is still there)
2. Correct the path to the item so a proper installs_array is generated.

Reply:
Coming back to this: I'm generally cool with this concept and code, but I still would like to see a more informative message here. The things you recommend as courses of action should be put into text somehow, or linked/documented on the wiki.

Generally speaking, I think AutoPkg has a lot of really inscrutable failure states and I don't want to create more of them. If there's something wrong that the admin has control over, we should try our best to tell them to succeed.

Reply:
so a PR with a wiki update?.. if the PR is merged into a release I could update the wiki post..